### DISCUSSION TITLE
Potential issue with Dolphin integration - "Not a shared directory"

### DISCUSSION THREAD

**Discussion by ghost** [Martchus/syncthingtray#274] [General]
The Dolphin plugin is not recognizing shared folders so the context menu actions are unavailable. All syncthing and syncthing tray functions otherwise work fine. Before I file a formal bug report, is there anything I could do to troubleshoot?

What I've tried so far:

1. Relogged
2. Rebooted
3. Created a new folder and shared it
4. Double checked/pasted API key in ~\\.config\syncthingfileitemaction.ini
5. Selected config.xml and config.xml.v0 in ~\\.local\state\syncthing (and again relogged/rebooted)

My config:
1. Linux 6.9.9-arch1-1 (Arch Linux)
2. KDE Plasma 6.1.2 (Wayland)
3. Syncthing v1.27.9, Syncthing Tray v1.5.5 

![issue](https://github.com/user-attachments/assets/e6a64a84-9574-4e6a-a24a-a02cb9938a7c)

**Comment by Martchus** [comment]
That's strange. I'm also using Arch and KDE/Plasma/Wayland and haven't encountered this issue yet. Your Syncthing config is also under a standard location so you should not even need to select it manually.

The plugin could at least be loaded. I suppose I don't know how to troubleshoot this except for doing a deep investigation with gdb.

**Reply by Martchus** [reply]
One thing you could try first: Launch Dolphin from a terminal. Maybe stderr/stdout contain useful information. You can also set [additional environment variables for logging](https://github.com/Martchus/syncthingtray?tab=readme-ov-file#logging).

Note that debugging the Dolphin integration (and KDE plugins in general) is a bit of a rabbit hole you might want to avoid. But if you dare, I have actually noted down [the steps to get started](https://github.com/Martchus/syncthingtray/blob/master/fileitemactionplugin/testing.md) (using Qt Creator).

**Reply by ghost** [reply]
Here's what I get in the terminal after right-clicking on a file (tried with two different files) and then on the directory itself and mousing over to the Syncthing context menu item each time. If nothing jumps out as relevant I'll see if I can get the time to dig a little deeper with Qt Creator. 

```
> kf.coreaddons: The plugin "/usr/lib/qt6/plugins/kf6/kfileitemaction/libsyncthingfileitemaction-qt6.so" explicitly states an Id in the embedded metadata, which is different from the one derived from the filename The Id field from the KPlugin object in the metadata should be removed
> 
> kf.purpose.externalprocess: Cannot initialize model for plugin type "Export" with data QJsonObject(): missing key "urls"
> 
> Info: Syncthing config loaded from "/home/oryschakj/.local/state/syncthing/config.xml"
> 
> QGestureManager::deliverEvent: could not find the target for gesture
> 
> qt.qpa.wayland: Creating a popup with a parent, QWidgetWindow(0x5ad1f1427c00, name="Dolphin#1Window") which does not match the current topmost grabbing popup, QWidgetWindow(0x5ad1f1d3fa00, name="DolphinContextMenuClassWindow") With some shell surface protocols, this is not allowed. The wayland QPA plugin is currently handling it by setting the parent to the topmost grabbing popup. Note, however, that this may cause positioning errors and popups closing unxpectedly. Please fix the transient parent of the popup.
> 
> kf.coreaddons: The plugin "/usr/lib/qt6/plugins/kf6/kfileitemaction/libsyncthingfileitemaction-qt6.so" explicitly states an Id in the embedded metadata, which is different from the one derived from the filename The Id field from the KPlugin object in the metadata should be removed
> 
> QGestureManager::deliverEvent: could not find the target for gesture
> 
> qt.qpa.wayland: Creating a popup with a parent, QWidgetWindow(0x5ad1f1427c00, name="Dolphin#1Window") which does not match the current topmost grabbing popup, QWidgetWindow(0x5ad1f1ae0920, name="DolphinContextMenuClassWindow") With some shell surface protocols, this is not allowed. The wayland QPA plugin is currently handling it by setting the parent to the topmost grabbing popup. Note, however, that this may cause positioning errors and popups closing unxpectedly. Please fix the transient parent of the popup.
```

**Reply by Martchus** [reply]
Unfortunately this is not much to work with. You should however see more with `export LIB_SYNCTHING_CONNECTOR_LOG_ALL=1`. Probably relevant logs also appear already directly when launching Dolphin (and not only after right-clicking).

**Reply by Martchus** [reply]
Maybe the use of `~` is the problem? I hardly use that Syncthing feature so its handling in the Dolphin integration is probably not well tested. (And there were issues with it before in other components like https://github.com/Martchus/syncthingtray/issues/148.)

**Reply by ghost** [reply]
> Maybe the use of `~` is the problem? I hardly use that Syncthing feature so its handling in the Dolphin integration is probably not well tested. (And there were issues with it before in other components like #148.)

Yep, that was it. I manually set the path in the local syncthing config and all the appropriate context menu items now pop up.

**Reply by Martchus** [reply]
I'll come up with a fix for `~` next week.

**Reply by ghost** [reply]
> I'll come up with a fix for `~` next week.

Thank you! I really appreciate your help

**Reply by Martchus** [reply]
Using `~` should now be supported by the Dolphin integration, see https://github.com/Martchus/syncthingtray/commit/f73053249a4dfd0e93eb6cf0b1e0d1b182aa3005.
